jetty-ee9/jetty-ee9-nested/src/main/java/org/eclipse/jetty/ee9/nested/Response.java

```diff
@@ -792,13 +792,16 @@ private String getCharacterEncoding(boolean setContentType)
         if (_mimeType != null && _mimeType.isCharsetAssumed())
             return _mimeType.getCharsetString();
 
+        // Get MimeTypes from context
+        MimeTypes mimeTypes = _channel.getRequest().getCoreRequest().getContext().getMimeTypes();
```

Comment:
That's a long way to go, isn't the Context available already from this line?

Reply:
No.  Need to get the context and it is only available from the request.